app/search/gleaner.py

```diff
@@ -28,48 +32,64 @@ def build_query(user_query="", page_number=1):
                 (GROUP_CONCAT(DISTINCT ?sameAs ; separator=", ") as ?sameAs)
                 (GROUP_CONCAT(DISTINCT ?keywords ; separator=", ") as ?keywords)
                 (GROUP_CONCAT(DISTINCT ?temporal_coverage ; separator=", ") as ?temporal_coverage)
+                (GROUP_CONCAT(DISTINCT ?spatial_coverage ; separator=", ") as ?spatial_coverage)
 
             {{
               VALUES ?type {{ schema:Dataset sschema:Dataset }}
               ?s a ?type .
               {{
                 ?s sschema:name ?title .
-                ?s sschema:keywords ?keywords .
-                ?s sschema:url ?url .
+                {{ ?s sschema:keywords ?keywords . }} UNION {{
+                    ?catalog ?relationship ?s .
+                    ?catalog sschema:keywords ?keywords .
+                }}
                 ?s sschema:description | sschema:description/sschema:value  ?abstract .
-                ?s sschema:identifier | sschema:identifier/sschema:value ?id .
                 ?s sschema:temporalCoverage ?temporal_coverage .
+                ?s sschema:spatialCoverage ?spatial_coverage .
 
                 OPTIONAL {{
                     ?s sschema:sameAs ?sameAs .
                 }}
                 OPTIONAL {{
                     ?s sschema:license ?license .
-                    
+                }}
+                OPTIONAL {{
+                    ?s sschema:url ?url .
```

Comment:
CCHDO datasets don't have a URL or an identifier specified! But they do have an `@id` in the json-ld metadata. Which is what the `BIND` is doing below.